server/graphql/v1/mutations/members.js

```diff
@@ -4,93 +4,80 @@ import models from '../../../models';
 import errors from '../../../lib/errors';
 import roles from '../../../constants/roles';
 
-export function createMember(_, args, req) {
-  let collective;
-
-  const checkPermission = () => {
+export async function createMember(_, args, req) {
+  const checkPermission = collective => {
     if (!req.remoteUser) {
       throw new errors.Unauthorized('You need to be logged in to create a member');
     }
-    if (req.remoteUser.isAdmin(collective.id)) {
-      return true;
+    if (!req.remoteUser.isAdmin(collective.id)) {
+      throw new errors.Unauthorized(
+        `You need to be logged in as a core contributor or as a host of the ${collective.slug} collective`,
+      );
     }
-    throw new errors.Unauthorized(
-      `You need to be logged in as a core contributor or as a host of the ${collective.slug} collective`,
-    );
   };
 
-  return (
-    req.loaders.collective.findById
-      .load(args.collective.id)
-      .then(c => {
-        if (!c) throw new Error(`Collective with id ${args.collective.id} not found`);
-        collective = c;
-      })
-      .then(() => {
-        if (args.role !== roles.FOLLOWER) {
-          return checkPermission();
-        } else {
-          return null;
-        }
-      })
-      // find or create user
-      .then(() => {
-        if (args.member.id) {
-          return req.loaders.collective.findById.load(args.member.id).then(memberCollective => {
-            return {
-              id: memberCollective.CreatedByUserId,
-              CollectiveId: memberCollective.id,
-            };
-          });
-        }
-      })
-      .then(u => u || models.User.findOrCreateByEmail(args.member.email, args.member))
-      // add user as member of the collective
-      .then(user =>
-        models.Member.create({
-          CreatedByUserId: user.id,
-          MemberCollectiveId: user.CollectiveId,
-          CollectiveId: collective.id,
-          role: args.role.toUpperCase() || roles.FOLLOWER,
-          since: args.since,
-        }),
-      )
-  );
-}
+  const collective = await req.loaders.collective.findById.load(args.collective.id);
+
+  if (!collective) {
+    throw new Error(`Collective with id ${args.collective.id} not found`);
+  }
+
+  if (args.role !== roles.FOLLOWER) {
+    checkPermission(collective);
+  }
 
-export function removeMember(_, args, req) {
-  let membership;
+  let user;
+  if (args.member.id) {
+    user = await models.User.findOne({ where: { CollectiveId: args.member.id } });
+  }
+  if (!user) {
+    user = await models.User.findOrCreateByEmail(args.member.email, args.member);
+  }
 
-  const checkPermission = () => {
+  return models.Member.create({
+    // NOTE: doesn't look like a good idea to set CreatedByUserId != req.remoteUser.id
```

Comment:
Agree, I was worried about this too. It seemed fine in the code I inspected though.